File: docker/pull.jl
dtags = Dict("c6"=>"centos6","c7"=>"centos7","u14"=>"ubuntu14","f22"=>"fedora22")
info("Available OS tags: ",join(keys(dtags),", "))
if length(ARGS) == 0 println("Usage error: Please provide Docker username as first argument.\n\tSpecify a subset of tags by listing them as additional arguments."); exit() end
duser = ARGS[1]
name="sim-recon"
for tag in keys(dtags); if length(ARGS) > 1 if !(tag in ARGS) continue end end
    repo = string(joinpath(duser,name),":",dtags[tag])
    run(`docker pull $repo`)
    try run(`docker rmi $name:$tag`)
    catch info("Image not available to remove (ignore previous error).") end
    run(`docker tag $repo $name:$tag`); run(`docker rmi $repo`)
end

File: docker/gxd/cmd/pull.go
package cmd

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"
)

// Create the pull command
var cmdPull = &cobra.Command{
	Use:   "pull [TAG...]",
	Short: "Pull docker images",
	Long: `Pull docker images.

tags: c6, c7, u16

All tags will be pulled if no arguments are given.`,
	Example: `1. gxd pull -u USER c6`,
	Run:     runPull,
}

func init() {
	cmdGXD.AddCommand(cmdPull)

	cmdPull.Flags().StringVarP(&USER, "user", "u", "", "Docker username")
	cmdPull.Flags().BoolVarP(&rmi, "rmi", "", false, "Remove old image before pull")
}

func runPull(cmd *cobra.Command, args []string) {
	if USER == "" {
		exitNoUsername(cmd)
	}
	var names = map[string]string{
		"c6":  "centos6",
		"c7":  "centos7",
		"u16": "ubuntu16",
	}

	var tags = []string{"c6", "c7", "u16"}

	for _, arg := range args {
		if !in(tags, arg) {
			fmt.Fprintf(os.Stderr, "%s: Unknown tag\n", arg)
			os.Exit(2)
		}
	}

	if len(args) == 0 {
		args = tags
	}

	name := "sim-recon-deps"
	for _, tag := range tags {
		if !in(args, tag) {
			continue
		}
		repo := USER + "/" + name + ":" + names[tag]
		run("docker", "pull", repo)
		if rmi {
			run("docker", "rmi", name+":"+tag)
		}
		run("docker", "tag", repo, name+":"+tag)
		run("docker", "rmi", repo)
	}
}
